### PR TITLE
integration of firebase for signup, login, forgot-password, header photo and edit profile popup

### PR DESCRIPTION
**Pull request by Manan-Sharma-5**
## Summary by CodeRabbit

- **New Features**
	- Introduced a password recovery feature allowing users to request a password reset via email.
	- Enhanced login functionality to accept either email or username for authentication.
	- Added user registration capabilities within the signup process.
	- Improved profile management with user data fetching and updating functionalities.
	- Streamlined Facebook login integration with a new library for better user experience.

- **Bug Fixes**
	- Resolved issues related to the previous password recovery and login processes.

- **Documentation**
	- Updated documentation to reflect new authentication methods and user profile management features.

- **Style**
	- Adjusted styling for better visual hierarchy in the profile edit interface.

### CODE REVIEW

**Comment by Manan-Sharma-5** [comment]
[Login through email and password](https://trello.com/c/8N6EBLnV/28-login-through-email-and-password)

**Comment by Manan-Sharma-5** [comment]
[Forgot Password to reset the password](https://trello.com/c/uwUAlJpz/29-forgot-password-to-reset-the-password)

**Comment by Manan-Sharma-5** [comment]
[Profile Information](https://trello.com/c/CHk2RgqE/30-profile-information)

**Comment by Manan-Sharma-5** [comment]
[Signup through email and password](https://trello.com/c/UhAYNK6K/27-signup-through-email-and-password)

**Comment by Manan-Sharma-5** [comment]
[react-facebook-login -> need to change the legacy packages](https://trello.com/c/o4DYExFo/54-react-facebook-login-need-to-change-the-legacy-packages)

File: components/profile/ProfileIcon.tsx [hunk 1]
```diff
@@ -2,26 +2,112 @@ import { View } from "@/components/Themed";
...
+        console.error("Error getting document:", error);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    if (user) {
+      fetchUserProfile();
+    }
+  }, [user]);
```
Comment (dev-rathore) [inline components/profile/ProfileIcon.tsx:64]: Should we shift this fetchUserProfile in user-context and we can fetch userProfile from there and utilize it wherever we want using the context?

File: contexts/auth-context.provider.tsx [hunk 2]
```diff
@@ -1,47 +1,192 @@
...
+            errorMessage = error.message;
+        }
+
+        Toaster.error("Sign Up Error", errorMessage);
+      });
+  };
+
+  const forgotPassword = async (email: string) => {
+    try {
+      //check from datastore collection
```
Comment (dev-rathore) [inline contexts/auth-context.provider.tsx:145]: If not required, we can remove this commented code to avoid confusion.

**Review by dev-rathore — APPROVED** [review]
Everything else looks good to me apart from two comments

**Pull request merged**